### PR TITLE
Build wheels

### PR DESCRIPTION
**Pull request by mgeplf**

### CODE REVIEW

File: packaging/python_wheel/setup.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env python
+from setuptools import setup, Distribution
+
+VERSION = '0.0.0'
```
Comment (mgeplf) [inline packaging/python_wheel/setup.py]: Need a good way to get version information, either:
 - through python
 - have cmake put it in this file

File: packaging/python_wheel/setup.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+class BinaryDistribution(Distribution):
+    def is_pure(self):
+        return False
+
+    def has_ext_modules(self):
+        return True
+
+
+setup(
+    name='brain',
```
Comment (mgeplf) [inline packaging/python_wheel/setup.py]: will need to add copyright, etc, info here

File: packaging/python_wheel/README.txt [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        -v $BASE:/io quay.io/pypa/manylinux1_x86_64 \
+        /bin/bash
+    $ cd /io/packaging/python_wheel
+
+Possible Improvements
+*********************
+ - A custom Dockerfile could be created so that CMake/HDF5 isn't rebuilt each
+   time, however, the time saving from this wouldn't be large, and potentially
+   complicates the deployment.
+ - Only recompile BoostPython, instead of all of Boost
```
Comment (tribal-tec) [inline packaging/python_wheel/README.txt]:
```
./b2 --build-type=minimal --with-python
```
might do the trick.

File: packaging/python_wheel/setup.py (hunk 2 above)
Comment (tribal-tec) [inline packaging/python_wheel/setup.py]: probably addressing this and the version info with a cmake pass

File: packaging/python_wheel/README.txt (hunk 3 above)
Comment (mgeplf) [inline packaging/python_wheel/README.txt]: I think --build-type=minimal is the default.  What I'd really like to do is see if I can install everything except boost-python, and then install boost-python for each python version, as the wheel needs to be built.

**Comment by mgeplf** [comment]
I think the version naming has to be done in this version before anything gets merged.

File: packaging/python_wheel/README.txt (hunk 3 above)
Comment (hernando) [inline packaging/python_wheel/README.txt]: That's possible, actually we have both python 2.6 and 3.4 in our boost module.

The code looks something like this:

    module load BBP/viz/python/3.4.3
    ./bootstrap.sh --with-python=python3.4 --prefix=...
    ./b2 --build-type=minimal --with-python install 

     module unload BBP/viz/python/3.4.3
     ./b2 -a
     ./bootstrap.sh --prefix=...
     ./b2 --build-type=minimal install

File: packaging/python_wheel/setup.py.in [hunk 4]
```diff
@@ -14,8 +11,12 @@ def has_ext_modules(self):
 
 setup(
     name='brain',
-    version=VERSION,
-    description='brain',
+    version='${Brion_VERSION}',
```
Comment (hernando) [inline packaging/python_wheel/setup.py.in]: So, the build process for the Python package will generate the brain/version.py module automatically or do we still need to create it?

Comment (mgeplf) [inline packaging/python_wheel/setup.py.in]: The build process doesn't currently create a version.py, which would probably make sense.

File: packaging/python_wheel/README.txt (hunk 3 above)
Comment (mgeplf) [inline packaging/python_wheel/README.txt]: Ok, interesting.  Probably the way to go.  I didn't know boost build system well enough to know if it's robust, but this seems to show that it is.  Thanks.

**Comment by mgeplf** [comment]
I'm good, but @hernando's suggestion wrt version.py should be done at some point.

**Comment by tribal-tec** [comment]
That's https://github.com/BlueBrain/Brion/issues/155 I guess? It's a separate thing to me.

**Comment by mgeplf** [comment]
It's related, and it's probably easier to setup using the cmake generation like you have now, but yeah, separate from you.

**Pull request merged**